### PR TITLE
support listing works with its labels

### PR DESCRIPTION
**Pull request by skeeey**

### CODE REVIEW

**Comment by skeeey** [comment]
should wait for #133 is merged

File: pkg/client/cloudevents/grpcsource/util.go
Comment (machi1990) [inline pkg/client/cloudevents/grpcsource/util.go:111]: With https://github.com/openshift-online/maestro/pull/136 being merged, can we check what other operators are supported and allow list them as valid operators? 

/cc @clyang82

Comment (skeeey) [reply pkg/client/cloudevents/grpcsource/util.go]: @machi1990 now only the `DoesNotExist` cannot be supported (if this is expected, we need enhance the `ocm-sdk` as a follow up), I added some test case in the e2e `List works with gRPC source ManifestWork client`, and I also add the support for label selector in the watch, PTAL, thanks

Comment (machi1990) [reply pkg/client/cloudevents/grpcsource/util.go]: Thanks @skeeey 

>  now only the DoesNotExist cannot be supported (if this is expected, we need enhance the ocm-sdk as a follow up)

That's noted. Just out of curiosity, wouldn't his be the opposite of what's done for Exists? At least that's my expectation but maybe I am missing something.

> I added some test case in the e2e List works with gRPC source ManifestWork client, and I also add the support for label selector in the watch, PTAL, thanks

I am giving it a look now. Thanks

Comment (machi1990) [reply pkg/client/cloudevents/grpcsource/util.go]: The change lgtm. 
Once this is merged, I can test it against my client I am building.

**Comment by machi1990** [comment]
lgtm

Once merged, I can test it against the client I am building.

File: pkg/client/cloudevents/grpcsource/util.go
Comment (clyang82) [inline pkg/client/cloudevents/grpcsource/util.go:135]: maybe extract `payload->'metadata'->'labels'` as a constant since it is using in many places.

File: pkg/client/cloudevents/grpcsource/util.go
Comment (clyang82) [inline pkg/client/cloudevents/grpcsource/util.go:172]: Does kube label selector support `or`?

Comment (skeeey) [reply pkg/client/cloudevents/grpcsource/util.go:172]: no, it does not

File: pkg/client/cloudevents/grpcsource/util.go
Comment (skeeey) [reply pkg/client/cloudevents/grpcsource/util.go]: done

**Review by clyang82 — APPROVED** [review]
LGTM

**Comment by machi1990** [comment]
> lgtm
> 
> Once merged, I can test it against the client I am building.

Just an FYI; I've tested the PR with a code similar to this 
```

l := &corev1.NamespaceList{}

err = client.List(ctx, l, k8sClient.MatchingLabels(map[string]string{
"key": "label",
}))

```

and works as expected. Thanks @skeeey @clyang82 @qiujian16

**Pull request merged**
